src/integrators/wrapper_integrator.hpp

```diff
@@ -156,6 +157,72 @@ class MixedBilinearToNonlinearFormIntegrator : public mfem::NonlinearFormIntegra
   std::shared_ptr<mfem::ParFiniteElementSpace> trial_fes_;
 };
 
+/**
+ * @brief A class to convert NonlinearFormIntegrator to one where the input parameter undergoes a change of variables
+ */
+class TransformedNonlinearFormIntegrator : public mfem::NonlinearFormIntegrator {
+public:
+  /// alias for prototype of the residual_func
+  using transformed_func = std::shared_ptr<mfem::Vector>(const mfem::FiniteElement& el, mfem::ElementTransformation& Tr,
+                                                         const mfem::Vector&);
+
+  /// alias for prototype of the gradient of residual_func
+  using transformed_gradient_func = std::shared_ptr<mfem::DenseMatrix>(const mfem::FiniteElement&   el,
+                                                                       mfem::ElementTransformation& Tr,
+                                                                       const mfem::DenseMatrix&);
+
```

Comment:
Do these need to be `shared_ptr`s, or can they just return `mfem::Vector`/`mfem::DenseMatrix`?

Reply:
I guess there are a few more options that I can think of:
1. I can mutate the `mfem::Vector/mfem::DenseMatrix` being passed in.
2. I could take in the output `mfem::Vector/mfem::DenseMatrix` by reference.
3. I could also return unique_ptr or raw ptr.